### PR TITLE
Fix #6236: drop @Repeated when inferring type args, return types

### LINKED ISSUE #6236: `@Repeated` can leak as method result type
We probably should drop the `@Repeated` annotation when inferring the result type of a method:

```scala
scala> def foo(s: String*) = s
def foo(s: String*): Seq[String] @Repeated

scala> def foo(s: String*) = { val s1 = s; s1 }
def foo(s: String*): Seq[String]
```

### PR DESCRIPTION
**Pull request by AleksanderBG**

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/core/Types.scala [hunk 1]
```diff
@@ -1485,6 +1485,13 @@ object Types {
      */
     def signature(implicit ctx: Context): Signature = Signature.NotAMethod
 
+    def dropRepeatedAnnot(implicit ctx: Context): Type = this match {
+      case AnnotatedType(parent, annot) if annot.symbol eq defn.RepeatedAnnot => parent
+      case tp @ AnnotatedType(parent, annot) =>
+        tp.derivedAnnotatedType(parent.dropRepeatedAnnot.stripPoly, annot)
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/core/Types.scala]: Why the .stripPoly ?

File: compiler/src/dotty/tools/dotc/typer/Namer.scala [hunk 2]
```diff
@@ -1392,7 +1392,7 @@ class Namer { typer: Typer =>
       case _ =>
         WildcardType
     }
-    paramFn(checkSimpleKinded(typedAheadType(mdef.tpt, tptProto)).tpe)
+    paramFn(checkSimpleKinded(typedAheadType(mdef.tpt, tptProto)).tpe.dropRepeatedAnnot)
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/typer/Namer.scala]: I would move this into `def widenRhs` above.

File: compiler/src/dotty/tools/dotc/core/Types.scala (hunk 1 above)
Comment (AleksanderBG) [reply compiler/src/dotty/tools/dotc/core/Types.scala]: That is a very good question and my answer is: I don't know! I must have accidentally pasted this code from somewhere when committing.

File: compiler/src/dotty/tools/dotc/typer/Namer.scala (hunk 2 above)
Comment (AleksanderBG) [reply compiler/src/dotty/tools/dotc/typer/Namer.scala]: Done.

File: compiler/src/dotty/tools/dotc/core/ConstraintHandling.scala [hunk 3]
```diff
@@ -302,7 +302,7 @@ trait ConstraintHandling[AbstractContext] {
       if (isMultiSingleton(tp) && !isMultiSingleton(bound) &&
           !isSubTypeWhenFrozen(bound, defn.SingletonType)) tp.widen
       else tp
-    widenOr(widenSingle(tp))
+    widenOr(widenSingle(tp)).dropRepeatedAnnot
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/core/ConstraintHandling.scala]: Update the method documentation.

File: compiler/src/dotty/tools/dotc/typer/Namer.scala [hunk 4]
```diff
@@ -1320,10 +1320,13 @@ class Namer { typer: Typer =>
...
-        case ctp: ConstantType if isInlineVal => ctp
-        case ref: TypeRef if ref.symbol.is(ModuleClass) => tp
-        case _ => tp.widen.widenUnion
+      def widenRhs(tp: Type): Type = {
+        val tp1 = tp.widenTermRefExpr match {
+          case ctp: ConstantType if isInlineVal => ctp
+          case ref: TypeRef if ref.symbol.is(ModuleClass) => tp
+          case _ => tp.widen.widenUnion
+        }
+        tp1.dropRepeatedAnnot
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/typer/Namer.scala]: Update the method documentation.

**Pull request merged**